File: 12. 整数转罗马数字/12. 整数转罗马数字.java
public class Solution {
    public String intToRoman(int num) {
        int[] nums = {1000,900,500, 400,100, 90 , 50,  40, 10,  9 , 5 ,  4 , 1 };
        String[] romans = {"M","CM","D","CD","C","XC","L","XL","X","IX","V","IV","I"};
        int i = 0;
        String res = "";

        while(num>0){
            if(nums[i]<=num){
                num -= nums[i];
                res += romans[i];
            }else{
                i++;
            }

        }
        
        return res;
    }
}

// 是按python版翻译的，python版这个方法挺不错的

File: 12. 整数转罗马数字/12. 整数转罗马数字.py
class Solution:
    def intToRoman(self, num: int) -> str:
        res = ''
        integ = [1000,900,500, 400,100, 90 , 50,  40, 10,  9 , 5 ,  4 , 1 ]
        roman = ['M','CM','D','CD','C','XC','L','XL','X','IX','V','IV','I']
        n = len(integ)
        i = 0
        while i < n:
            if num>=integ[i]:
                res += roman[i]
                num -= integ[i]
            else:
                i += 1
        return res

s = Solution()
t = s.intToRoman(1994)
print(t)    